src/types.ts

```diff
@@ -0,0 +1,12 @@
+/***************************************************************************************************
+ *  Copyright (c) Red Hat, Inc. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE file in the project root for license information.
+ **************************************************************************************************/
+
+type CommandResult = {
+    exitCode: number
+    output: string
+    error: string
+};
+
+export default CommandResult;
```

Comment:
This is not a good use of the default export syntax. If you want to add another exported type to this file, there is no good way to do that without breaking all files that import the CommandResult.

You can either namespace it:
```
namespace Types {
  export type CommandResult {
     <etc>
  }
}
export default Types;
```
You can't do `export namespace Types`, I'm not sure why exactly. But you can achieve the same effect by exporting it after its declaration and this has the added bonus of being able to merge the namespace with another type if you choose. https://www.typescriptlang.org/docs/handbook/declaration-merging.html

then users can `import Types from "./types"` and reference `Types.CommandResult`.

or in this case, when it's "generic" or "utility" types or functions that I don't care to namespace. 
I prefer to do it like this:
```
export type CommandResult = {
  <etc>
}
// no default export
```
then users do
```
import { CommandResult } from "./types"
```

and any new types are just added to that import statement.

I would use the default export if I had no intention of ever adding another export to this file, but that is not the case here with a generic `types.ts`.

Reply:
Yeah, second approach seems to be better.
Thanks for the detailed explanation, this is very helpful.